react/Components/inputButtonArea.tsx

```diff
@@ -15,39 +16,90 @@ const InputButtonArea: FC = () => {
     code,
     updateGiftCardFunction,
     setAddValueGiftCard,
+    copyCode,
+    setValidation,
+    loading,
+    loadingCode,
     addValueGiftCard,
   } = useStore()
 
-  function submitFunctionValueButton(e: React.FormEvent<HTMLFormElement>) {
+  function submitFunctionValueButton(e: { preventDefault: () => void }) {
     e.preventDefault()
     updateGiftCardFunction()
   }
 
   function submitFunctionCodeButton(e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault()
-    if (!navigator.clipboard) return
-    navigator.clipboard.writeText(code)
+    copyCode()
+  }
+
+  const currencyConfig = {
+    locale: 'pt-BR',
+    formats: {
+      number: {
+        BRL: {
+          style: 'currency',
+          currency: 'BRL',
+          minimumFractionDigits: 2,
+          maximumFractionDigits: 2,
+        },
+      },
+    },
+  }
+
+  const handleChange = (
+    event: { preventDefault: () => void },
```

Comment:
I'm not sure if my suggestion works well, but I think it's worth a try 😊 
```suggestion
    event: FocusEvent<HTMLInputElement>,
```

Reply:
Don't works with FocusEvent<HTMLInputElement> , but works with FocusEvent, it's the same thing? 🤔

Reply:
*Correction:
Don't works with  FocusEvent<HTMLInputElement> , but works with FocusEvent, it's the same thing? 🤔

Reply:
Worked with what `Event`?

Reply:
I solved!!!
I add React. before FocusEvent and works
Thanks! :)